infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -546,92 +568,48 @@ public void updateSyncStatus(String clusterId, SyncType lastSyncType, SyncStatus
+                log.info("syncStatusPoller :: skip replication status update, log entry replication is {}",
+                        previous.getStatus());
+                txn.commit();
                 return;
-            }
-
-            if (snapshotStatus == null){
-                log.warn("syncStatusPoller [logEntry]:: previous snapshot status is not present for cluster: {}", clusterId);
-                snapshotStatus = SnapshotSyncInfo.newBuilder().build();
-            }
-
-            current = ReplicationStatusVal.newBuilder()
-                    .setRemainingEntriesToSend(remainingEntries)
-                    .setSyncType(type)
-                    .setStatus(SyncStatus.ONGOING)
-                    .setSnapshotSyncInfo(snapshotStatus)
-                    .build();
-
-            try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
-                txn.putRecord(replicationStatusTable, key, current, null);
+            } else if (type == SyncType.SNAPSHOT && (previousSnapshotSyncInfo.getStatus().equals(SyncStatus.NOT_STARTED)
+                    || previousSnapshotSyncInfo.getStatus().equals(SyncStatus.STOPPED))) {
+                // Skip update of sync status, it will be updated once replication is resumed or started
+                log.info("syncStatusPoller :: skip replication status update, snapshot sync is {}",
+                        previousSnapshotSyncInfo.getStatus());
                 txn.commit();
-            }
-            
-            log.debug("syncStatusPoller :: Log Entry status set to ONGOING, clusterId: {}, remainingEntries: {}, " +
-                            "snapshotSyncInfo: {}", clusterId, remainingEntries, snapshotStatus);
-        } else if (type == SyncType.SNAPSHOT) {
-
-            SnapshotSyncInfo currentSnapshotSyncInfo;
-            if (snapshotStatus == null){
-                log.warn("syncStatusPoller [snapshot] :: previous status is not present for cluster: {}", clusterId);
-                currentSnapshotSyncInfo = SnapshotSyncInfo.newBuilder().build();
-            } else {
-
-                if (snapshotStatus.getStatus().equals(SyncStatus.NOT_STARTED)
-                        || snapshotStatus.getStatus().equals(SyncStatus.STOPPED)) {
-                    // Skip update of sync status, it will be updated once replication is resumed or started
-                    log.info("syncStatusPoller :: skip replication status update, snapshot sync is {}", snapshotStatus);
-                    return;
-                }
-
-                currentSnapshotSyncInfo = snapshotStatus.toBuilder()
-                        .setStatus(SyncStatus.ONGOING)
-                        .build();
+                return;
             }
 
-            current = ReplicationStatusVal.newBuilder()
+            ReplicationStatusVal current = previous.toBuilder()
                     .setRemainingEntriesToSend(remainingEntries)
```

Comment:
the status also should be set to ONGOING, right?

Reply:
We could do it here, but it is also done on the entry of InLogEntrySyncState and InSnapshotSyncState so I decided to leave it out so the only thing the polling task modifies is the number of remaining entries.